src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -283,3 +306,37 @@ def find_version_info_in_manifest_file(
             patch_name = version_info[patch_field]
 
     return version_name, patch_name
+
+
+def project_path_to_deploy_path(
+    project_paths: List[Path], files_mapping: DeployMapping
+) -> List[Path]:
+    """Given a list of source paths, returns the deploy destination paths. This function assumes that a build was performed before calling it."""
```

Comment:
I'm verifying that the destination result exists on the filesystem, so the bundle existence is necessary. I could verify it outside of this function and have the coupling somewhere else. Not sure if it will be clearer.